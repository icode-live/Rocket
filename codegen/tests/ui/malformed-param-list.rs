#![feature(plugin, decl_macro)]
#![plugin(rocket_codegen)]

#[get("/><")]
fn get() -> &'static str { "hi" }

#[get("/<id><")]
fn get1(id: usize) -> &'static str { "hi" }

#[get("/<<<<id><")]
fn get2(id: usize) -> &'static str { "hi" }

#[get("/<!>")]
fn get3() -> &'static str { "hi" }

#[get("/<_>")]
fn get4() -> &'static str { "hi" }

#[get("/<1>")]
fn get5() -> &'static str { "hi" }

#[get("/<>name><")]
fn get6() -> &'static str { "hi" }

#[get("/<name>:<id>")]
fn get7() -> &'static str { "hi" }

#[get("/<>")]
fn get8() -> &'static str { "hi" }
