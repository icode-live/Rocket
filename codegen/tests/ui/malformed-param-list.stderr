error: malformed parameter
 --> $DIR/malformed-param-list.rs:4:9
  |
4 | #[get("/><")]
  |         ^^
  |
  = help: parameters must be of the form '<param>'

error: malformed parameter
 --> $DIR/malformed-param-list.rs:7:9
  |
7 | #[get("/<id><")]
  |         ^^^^^
  |
  = help: parameters must be of the form '<param>'

error: malformed parameter
  --> $DIR/malformed-param-list.rs:10:9
   |
10 | #[get("/<<<<id><")]
   |         ^^^^^^^^
   |
   = help: parameters must be of the form '<param>'

error: parameter names must be valid identifiers
  --> $DIR/malformed-param-list.rs:13:9
   |
13 | #[get("/<!>")]
   |         ^^^
   |
   = note: "!" is not a valid identifier

error: parameters must be named
  --> $DIR/malformed-param-list.rs:16:9
   |
16 | #[get("/<_>")]
   |         ^^^
   |
   = help: use a name such as `_guard` or `_param`

error: parameter names must be valid identifiers
  --> $DIR/malformed-param-list.rs:19:9
   |
19 | #[get("/<1>")]
   |         ^^^
   |
   = note: "1" is not a valid identifier

error: malformed parameter
  --> $DIR/malformed-param-list.rs:22:9
   |
22 | #[get("/<>name><")]
   |         ^^^^^^^^
   |
   = help: parameters must be of the form '<param>'

error: parameter names must be valid identifiers
  --> $DIR/malformed-param-list.rs:25:9
   |
25 | #[get("/<name>:<id>")]
   |         ^^^^^^^^^^^
   |
   = note: "name>:<id" is not a valid identifier

error: parameters cannot be empty
  --> $DIR/malformed-param-list.rs:28:9
   |
28 | #[get("/<>")]
   |         ^^

error: aborting due to 9 previous errors

